domains/pallets/auto-id/src/lib.rs

```diff
@@ -208,17 +207,21 @@ pub enum CertificateActionType {
 /// Signing data used to verify the certificate action.
 #[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
 pub struct CertificateAction {
+    /// On which AutoId the action is taken.
     pub id: Identifier,
+    /// Current nonce of the certificate.
     pub nonce: U256,
+    /// Type of action taken.
     pub action_type: CertificateActionType,
 }
 
 #[frame_support::pallet]
 mod pallet {
-    use crate::{AutoId, Identifier, RegisterAutoId, Signature};
+    use crate::{AutoId, Identifier, RegisterAutoId, Serial, Signature};
     use frame_support::pallet_prelude::*;
     use frame_support::traits::Time;
     use frame_system::pallet_prelude::*;
+    use scale_info::prelude::collections::BTreeSet;
```

Comment:
we are actively moving towards using `alloc` for no_std, can you use that instead ?